src/bootstrap/src/core/build_steps/vendor.rs

```diff
@@ -57,7 +57,9 @@ impl Step for Vendor {
         }
 
         // These submodules must be present for `x vendor` to work.
-        for submodule in SUBMODULES_FOR_RUSTBOOK.iter().chain(["src/tools/cargo"].iter()) {
+        for submodule in
```

Comment:
We already have `default_paths_to_vendor` (used below). Could we reuse that instead of hardcoding a path here?

Reply:
I don't think we can necessarily remove the hard-coded paths for the submodules. I'm assuming your intent is to try to avoid these two getting out of sync? I tried moving them up into `default_paths_to_vendor` in d8e8fc5575ba3f3aa31cb8c7f47df73e06a38e92 to go in that direction.

Reply:
Yes, sorry, that's what I meant, to make it a bit harder to avoid them going out of sync. This is much nicer, thank you!